src/shared/src/types/project-types.ts

```diff
@@ -75,3 +72,30 @@ export interface DescriptionBullet {
   dateDeleted?: Date;
   userChecked?: User;
 }
+
+export interface LinkType {
+  name: string;
+  dateCreated: Date;
+  creator: UserPreview;
+  iconName: string;
+}
+
+export interface Link {
+  linkId: string;
+  linkType: LinkType;
+  dateCreated: Date;
+  creator: User;
+  url: string;
+}
+
+export interface LinkCreateArgs {
+  linkId: string;
+  linkTypeName: string;
+  url: string;
+}
+
+export enum RequiredLinkType {
```

Comment:
required should just be a boolean that can be toggled on a linkType, having it as an enum here is weird

Reply:
mm thats smart actually